src/uproot/_dask.py

```diff
@@ -168,6 +170,162 @@ def dask(
+    args=None,
+    label=None,
+    token=None,
+    **kwargs,
+):
+    dask, da = uproot.extras.dask()
+    if not callable(func):
+        raise ValueError("`func` argument must be `callable`")
+    lengths = set()
+    iters = list(iterables)
+    for i, iterable in enumerate(iters):
+        if not isinstance(iterable, Iterable):
+            raise ValueError(
+                f"All elements of `iterables` must be Iterable, got {type(iterable)}"
+            )
+        try:
+            lengths.add(len(iterable))  # type: ignore
+        except (AttributeError, TypeError):
+            iters[i] = list(iterable)
+            lengths.add(len(iters[i]))  # type: ignore
+    if len(lengths) == 0:
+        raise ValueError("`from_map` requires at least one Iterable input")
+    elif len(lengths) > 1:
+        raise ValueError("All `iterables` must have the same length")
+    if lengths == {0}:
+        raise ValueError("All `iterables` must have a non-zero length")
+
+    # Check for `produces_tasks` and `creation_info`
+    produces_tasks = kwargs.pop("produces_tasks", False)
+    # creation_info = kwargs.pop("creation_info", None)
+
+    if produces_tasks or len(iters) == 1:
+        if len(iters) > 1:
+            # Tasks are not detected correctly when they are "packed"
+            # within an outer list/tuple
+            raise ValueError(
+                "Multiple iterables not supported when produces_tasks=True"
+            )
+        inputs = list(iters[0])
+        packed = False
+    else:
+        # Structure inputs such that the tuple of arguments pair each 0th,
+        # 1st, 2nd, ... elements together; for example:
+        # from_map(f, [1, 2, 3], [4, 5, 6]) --> [f(1, 4), f(2, 5), f(3, 6)]
+        inputs = list(zip(*iters))
+        packed = True
+
+    # Define collection name
+    label = label or dask.utils.funcname(func)
+    token = token or dask.base.tokenize(func, iters, **kwargs)
+    name = f"{label}-{token}"
+
+    # Define io_func
+    if packed or args or kwargs:
+        io_func = _PackedArgCallable(
+            func,
+            args=args,
+            kwargs=kwargs,
+            packed=packed,
+        )
+    else:
+        io_func = func
+
+    io_arg_map = dask.blockwise.BlockwiseDepDict(
+        mapping=LazyInputsDict(inputs),  # type: ignore
+        produces_tasks=produces_tasks,
+    )
+
+    dsk = dask.blockwise.Blockwise(
+        output=name,
+        output_indices="i",
+        dsk={name: (io_func, dask.blockwise.blockwise_token(0))},
+        indices=[(io_arg_map, "i")],
+        numblocks={},
+        annotations=None,
+    )
+
+    hlg = dask.highlevelgraph.HighLevelGraph.from_collections(name, dsk)
+    return dask.array.core.Array(hlg, name, chunks, dtype=dtype)
+
+
+class _UprootReadNumpy:
+    def __init__(self, hasbranches, key) -> None:
+        self.hasbranches = hasbranches
+        self.key = key
+
+    def __call__(self, i_start_stop):
+        i, start, stop = i_start_stop
```

Comment:
This makes me pine for Python 2's [tuple parameter unpacking](https://peps.python.org/pep-3113/)!